### DISCUSSION TITLE
Is it possible to get instrument name from Track structure?

### DISCUSSION THREAD

**Discussion by TheLucifurry** [CoderLine/alphaTab#918] [Q&A] [answered]
That's the whole question)

I've tried to parse it from `track.staves[0].stringTuning.name || track.name`, but this is unsafe way
Maybe, it'll be important.. Now, i set the instruments to tracks in TuxGuitar, than save it as gp5 and open via AlphaTab

**Comment by mxAlexG** [comment, marked as answer]
One way I know of is to get the midi program number in track.playbackInfo.program which you can write a function that resolves that to a string. Bare in mind that midi has several program numbers for guitars (overdrive, distortion, clean, etc).
From wiki:
[Guitar](https://en.wikipedia.org/wiki/Guitar)
25 [Acoustic Guitar (nylon)](https://en.wikipedia.org/wiki/Classical_guitar)
26 [Acoustic Guitar (steel)](https://en.wikipedia.org/wiki/Steel-string_acoustic_guitar)
27 [Electric Guitar (jazz)](https://en.wikipedia.org/wiki/Jazz_guitar)
28 [Electric Guitar](https://en.wikipedia.org/wiki/Electric_guitar) (clean)
29 Electric Guitar ([muted](https://en.wikipedia.org/wiki/Palm_mute))
30 Electric Guitar ([overdriven](https://en.wikipedia.org/wiki/Fuzzbox))
31 Electric Guitar ([distortion](https://en.wikipedia.org/wiki/Fuzzbox))
32 Electric Guitar ([harmonics](https://en.wikipedia.org/wiki/String_harmonic#Pinch_harmonics))

**Reply by TheLucifurry** [reply]
Oh, thank you, that will come in handy
~~I think, that we need a legal way to identify instrument from alphaTab tracks)~~

**Comment by Danielku15** [comment]
As @mxAlexG indicated: The best way to identify the instrument is the `track.playbackInfo.program`, this is the midi instrument used for playback. But you might need special treatment of the drum/percussion tracks where you might need to consider `staff.isPercussion`. 

Also see 
https://github.com/CoderLine/alphaTab/issues/358
https://github.com/CoderLine/alphaTab/issues/374

**Reply by TheLucifurry** [reply]
Thx!
